### PR TITLE
add sitemap file.

### PR DESCRIPTION
**Pull request by vwxyzh**
#240

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 1]
```diff
@@ -107,6 +107,16 @@ public void Build(DocumentBuildParameters parameters)
                         UpdateHref(manifest, context);
 
                         var generatedManifest = TemplateProcessor.Transform(processor, manifest.Select(s => s.Item).ToList(), context, parameters.ApplyTemplateSettings);
+
+                        // export sitemap
+                        YamlUtility.Serialize(
+                            Path.Combine(parameters.OutputBaseDir, "site.xref"),
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: "site.xref" move to constant? or make it configurable?

Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: Should we add file extension `yml` to it?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 2]
```diff
@@ -107,6 +107,16 @@ public void Build(DocumentBuildParameters parameters)
...
+                            from xref in context.XRefSpecMap.Values
+                            select new XRefSpec(xref)
+                            {
+                                Href = ((RelativePath)context.FileMap[xref.Href]).RemoveWorkingFolder().ToString() + "#" + XrefDetails.GetHtmlId(xref.Uid),
+                            });
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: extract to a data model? And from the experience of Manifest file/TOC, a object model is more extensible than an array model....

Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: what about other properties, e.g. Name, FullName...

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 3]
```diff
@@ -107,6 +107,16 @@ public void Build(DocumentBuildParameters parameters)
...
+
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: Log info when xref file is generated?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs (hunk 2 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: other properties is copied from old instance.

**Pull request closed without merging**